File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/Task.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import at.ac.tuwien.sepm.groupphase.backend.types.TaskStatus;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@EqualsAndHashCode
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private LocalDateTime startTime;

    @Column(nullable = false)
    private LocalDateTime endTime;

    @ManyToOne(fetch=FetchType.EAGER)
    private Employee assignedEmployee;

    @Column(nullable = false)
    private TaskStatus status;
}

File: frontend/src/app/dtos/task.ts
export class Task {
  constructor(public id: number,
              public title: string,
              public description: string,
              public startTime: string,
              public endTime: string,
              public assignedEmployeeUsername: string,
              public status: string,
              public subjectId: string,
              public subjectName: string,
              public isAnimalTask: boolean,
              public priority: boolean = false) {
  }
}
